ngraph/core/reference/include/ngraph/runtime/reference/sum.hpp

```diff
@@ -41,34 +42,32 @@ namespace ngraph
             }
 
             template <typename T>
-            void sum(const T* arg,
-                     T* out,
-                     const Shape& in_shape,
-                     const AxisSet& reduction_axes,
-                     bool keep_dims)
+            void sum(const T* arg, T* out, const Shape& in_shape, const AxisSet& reduction_axes)
             {
-                auto out_shape = reduce(in_shape, reduction_axes, keep_dims);
-                CoordinateTransform output_transform(out_shape);
-                std::vector<T> cs(shape_size(out_shape));
+                auto out_shape = reduce(in_shape, reduction_axes, false);
 
-                for (const Coordinate& output_coord : output_transform)
-                {
-                    out[output_transform.index(output_coord)] = 0;
-                    cs[output_transform.index(output_coord)] = 0;
-                }
+                std::vector<T> cs(shape_size(out_shape), 0);
+                std::fill(out, out + shape_size(out_shape), 0);
 
-                CoordinateTransform input_transform(in_shape);
+                const auto in_strides = row_major_strides(in_shape);
+                const auto out_strides = row_major_strides(out_shape);
 
+                CoordinateTransform input_transform(in_shape);
```

Comment:
Can we avoid `CoordinateTransform`?

Reply:
We actually need coordinates from input. To calculate `out_idx` we need input coordinate, then remove the reduces axes and then use strides to calculate the index in the tensor. Without it, we don't have a way to obtain which output index would correspond to input index. Additionally having `input_transform` object does not penalize performance, but the call to index() member function.